src/backend/marsha/core/permissions.py

```diff
@@ -1,20 +1,21 @@
 """Custom permission classes for the Marsha project."""
-from rest_framework import exceptions, permissions
+from rest_framework import permissions
 from rest_framework_simplejwt.models import TokenUser
 
-from .models.account import INSTRUCTOR, LTI_ROLES
+from .models.account import ADMINISTRATOR, INSTRUCTOR, LTI_ROLES
 
 
-class IsVideoInstructorTokenOrAdminUser(permissions.IsAdminUser):
-    """A custom permission class for JWT Tokens related to a video object.
+class BaseResourcePermission(permissions.IsAdminUser):
+    """Base permission class for JWT Tokens related to a resource object.
 
     These permissions build on the `IsAdminUser` class but grants additional specific accesses
-    to users authenticated with a JWT token built from a video ie related to a TokenUser as
+    to users authenticated with a JWT token built from a resource ie related to a TokenUser as
     defined in `rest_framework_simplejwt`.
 
     """
 
     message = "Only admin users or object owners are allowed."
```

Comment:
Funny, I couldn't find where this message is used...

Reply:
DRF does not use it when you combine multiple permissions. Should we keep it or not ?

Reply:
Actually I didn't find it in DRF's code base at all :man_shrugging:

Reply:
> Custom permissions will raise a PermissionDenied exception if the test fails. To change the error message associated with the exception, implement a message attribute directly on your custom permission. Otherwise the default_detail attribute from PermissionDenied will be used.

In the documentation

Reply:
I'm a bit lost but it seems your test below proves that this message is ignored and the `default_detail` is displayed from here: https://github.com/encode/django-rest-framework/blob/335054a5d36b352a58286b303b608b6bf48152f8/rest_framework/exceptions.py#L177

Reply:
Yes it's the case, the message is ignore because we are using the `or` operator between two permissions. If one fail the permission can't know wich message to use so it fallback to the default one.

Reply:
Ok! Makes sense. In any case the precised message should be set on each permission class and not on the base class. So I propose to remove it and keep the default message everywhere...

Reply:
![](https://media.giphy.com/media/KlFdtnpxEeH2E/giphy-downsized.gif)